Error copying file "gle.cpp" to "gle_cpp_lib.cpp".

While working on setting up GitHub Actions CI for GLE (#25) I found this build failure:

```
gmake[2]: Entering directory '/home/runner/work/GLE/GLE/build'
gmake[2]: Entering directory '/home/runner/work/GLE/GLE/build'
[ 30%] Generating gle_cpp_lib.cpp from gle.cpp
[ 31%] Generating gle_cpp_lib.cpp from gle.cpp
cd /home/runner/work/GLE/GLE/src/gle && /usr/local/bin/cmake -E copy gle.cpp gle_cpp_lib.cpp
gmake[2]: Leaving directory '/home/runner/work/GLE/GLE/build'
cd /home/runner/work/GLE/GLE/src/gle && /usr/local/bin/cmake -E copy gle.cpp gle_cpp_lib.cpp
/usr/bin/gmake  -f gle/CMakeFiles/gle.dir/build.make gle/CMakeFiles/gle.dir/build
Error copying file "gle.cpp" to "gle_cpp_lib.cpp".
gmake[2]: Entering directory '/home/runner/work/GLE/GLE/build'
gmake[2]: *** [gle/CMakeFiles/libgle-graphics_s.dir/build.make:74: /home/runner/work/GLE/GLE/src/gle/gle_cpp_lib.cpp] Error 1
gmake[2]: Leaving directory '/home/runner/work/GLE/GLE/build'
cd /home/runner/work/GLE/GLE/build && /usr/local/bin/cmake -E cmake_depends "Unix Makefiles" /home/runner/work/GLE/GLE/src /home/runner/work/GLE/GLE/src/gle /home/runner/work/GLE/GLE/build /home/runner/work/GLE/GLE/build/gle /home/runner/work/GLE/GLE/build/gle/CMakeFiles/libgle-graphics.dir/DependInfo.cmake "--color="
gmake[2]: *** Deleting file '/home/runner/work/GLE/GLE/src/gle/gle_cpp_lib.cpp'
gmake[1]: *** [CMakeFiles/Makefile2:270: gle/CMakeFiles/libgle-graphics_s.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
```

I am doing a parallel build with `-j4`.

It's trying to generate gle_cpp_lib.cpp from gle.cpp twice at the same time—likely once for the static library and once for the shared library—and one of them fails.

https://github.com/vlabella/GLE/blob/97cf2ca024ef9a12abf6fe6024439faf0c82a0e8/src/gle/CMakeLists.txt#L100-L101

This suggests dependencies aren't declared properly. I think this may be because you've only used `add_custom_command` but haven't used `add_custom_target` and then used that target as a dependency of whatever needs it.

https://github.com/vlabella/GLE/blob/97cf2ca024ef9a12abf6fe6024439faf0c82a0e8/src/gle/CMakeLists.txt#L81-L87

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -50,10 +50,6 @@ jobs:
             jobs=1
         esac
 
-        # Remove this jobs=1 override once parallel building is fixed.
-        # See https://github.com/vlabella/GLE/issues/26
-        jobs=1
-
         echo "--parallel $jobs" >> build.args
     - name: Configure
       run: |

--- a/src/gle/CMakeLists.txt
+++ b/src/gle/CMakeLists.txt
@@ -79,14 +79,6 @@ set( GLEBTOOL_SOURCES
 )
 
 add_custom_command(
- 	OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/gle_cpp_lib.cpp
- 	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/gle.cpp
- 	COMMAND ${CMAKE_COMMAND} -E copy gle.cpp gle_cpp_lib.cpp
-  	COMMENT "Generating gle_cpp_lib.cpp from gle.cpp"
-  	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-)
-
-add_custom_command(
  	OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/glerc
  	COMMAND ${CMAKE_COMMAND} -E echo begin config gle > ${CMAKE_CURRENT_BINARY_DIR}/glerc
  	COMMAND ${CMAKE_COMMAND} -E echo current = ${PROJECT_VERSION} >> ${CMAKE_CURRENT_BINARY_DIR}/glerc
@@ -97,10 +89,11 @@ add_custom_command(
 
 add_library(gle_common OBJECT ${GLE_SOURCES})
 add_executable( gle $<TARGET_OBJECTS:gle_common> gle.cpp )
-add_library( gle-graphics_s STATIC $<TARGET_OBJECTS:gle_common> gle_cpp_lib.cpp )
-add_library( gle-graphics SHARED $<TARGET_OBJECTS:gle_common> gle_cpp_lib.cpp )
+add_library(gle-graphics_s STATIC $<TARGET_OBJECTS:gle_common> gle.cpp)
+add_library(gle-graphics SHARED $<TARGET_OBJECTS:gle_common> gle.cpp)
+target_compile_definitions(gle-graphics_s PRIVATE -DHAVE_LIBGLE)
+target_compile_definitions(gle-graphics PRIVATE -DHAVE_LIBGLE)
 add_custom_target( glerc_file ALL DEPENDS glerc )
-set_source_files_properties( gle_cpp_lib.cpp PROPERTIES COMPILE_FLAGS -DHAVE_LIBGLE)
 
 set_target_properties(gle PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
 set_target_properties(gle-graphics_s PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})

--- a/src/gle/gle.cpp
+++ b/src/gle/gle.cpp
@@ -112,7 +112,7 @@ void gle_cat_csv(vector<string>* files) {
 	}
 }
 
-#if defined(HAVE_LIBGLE) || defined(HAVE_LIBGLE_STATIC)
+#ifdef HAVE_LIBGLE
 int GLEMain(int argc, char **argv) {
 #else
 int main(int argc, char **argv) {

